ws-client/src/client.rs

```diff
@@ -157,6 +160,8 @@ impl WsClient {
 			.await
 			.map_err(|e| Error::TransportError(Box::new(e)))?;
 
+		let (shutdown_sender, shutdown_receiver) = async_std::channel::unbounded::<()>();
```

Comment:
this is quite neat but no need for an unbounded channel, I think `Arc<AtomicBool>` would be simple enough for this usecase.

an alternative is to maybe put `error` in a `RwLock` then read it in `fn is_connected()`